packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -482,7 +493,7 @@ export class GarbageCollector implements IGarbageCollector {
          * may be out of date. This is fine because the state is updated every time GC runs based on the time then.
          */
         this.initializeBaseStateP = new LazyPromise<void>(async () => {
-            const currentReferenceTimestampMs = this.getCurrentReferenceTimestampMs();
+            const currentReferenceTimestampMs = this.runtime.getCurrentReferenceTimestampMs();
```

Comment:
>currentReferenceTimestampMs [](http://example.com/codeflow?start=18&length=27)

what happens if this   currentReferenceTimestampMs === undefined ? do we want to return or continue ?

Reply:
The code is written to handle if this is undefined. Take a look at `UnreferencedStateTracker` and what it does if current timestamp is undefined. Basically, we mark the node as unreferenced but do not track when it became unreferenced. Also, we cannot determine if a node can be deleted or not.
There have been some discussion or this topic and a proposed solution here - https://github.com/microsoft/FluidFramework/issues/9841